make LaTeX work via TinyTeX

rolandlo opened the pull request:
# Pull Request

## Description

This PR makes the LaTeX tool in Xournal++ work. It replaces the installation of `texlive-extra` by TinyTeX, which is a portable version, a subset of the `TeXLive` distribution, that is aimed at using less space.

Fixes #3

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

**Test Configuration**:

- Build the PR via `snapcraft` and install it
- remove the file `~/snap/xournalpp/current/.config/xournalpp/settings.xml` or modify the line
 ``` 
<property name="latexSettings.globalTemplatePath" value="/snap/xournalpp/x10/usr/share/xournalpp/resources/default_template.tex"/>
```  
in it to refer to the current version (replace `x10` by `current` in the example)
Alternatively update the global LaTeX template path via the GUI in Xournal++ (menu `Edit > Preferences`, LaTeX panel)

- `snap run xournalpp`
- Press the LaTeX Tool button 
![grafik](https://user-images.githubusercontent.com/40485433/128021331-5929e540-58a1-4d53-8616-99517a58e186.png)
and observe that the tool works and the formula can be modified
![grafik](https://user-images.githubusercontent.com/40485433/128019918-72613c02-d1b7-43d5-aa42-1f2a6738f613.png)

* OS: Ubuntu 21.04

**Additional information**:
- It is easy to add more LaTeX packages in the `snapcraft.yaml` if required (via the `tlmgr install` command)
- It would be possible to use the `TinyTeX-0` (bare minimum) release instead of `TinyTeX-1` and enrich it with hand-picked packages
- It would also be possible to use the bigger `TinyTeX` release (and enrich it with some hand-picked packages)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

rolandlo committed:
make LaTeX work via TinyTeX

kenvandine reviewed (commented):
How hard would it be to build TinyTex from source rather than fetch the binaries?  The binaries are just x86_64, I'd really love to have xournalpp published for arm64 in the future as well.

rolandlo commented:
> How hard would it be to build TinyTex from source rather than fetch the binaries? The binaries are just x86_64, I'd really love to have xournalpp published for arm64 in the future as well.

It would probably be easy, since there is also an [installer script](https://github.com/yihui/tinytex/blob/master/tools/install-unx.sh) for building from source in the TinyTeX Github repo. It was just not linked on the [main page](https://yihui.org/tinytex/). I tried it out locally in meanwhile. It was quite fast.

I simply did not think about arm64 and thought that the binaries might be be more reliable. I can modify the PR to make it build from source.

rolandlo committed:
build TinyTeX from source, customize  package list

The packages standalone, varwidth and scontents are needed for the default
LaTeX preamble in Xournal++. The packages pgf, gfplots are used for
creating plots and graphics. The listings package is used for code
highlighting and the xargs package for defining LaTeX  commands
with multiple arguments. The other packages are those from TinyTeX.

rolandlo commented:
I have made TinyTeX build from source and added a custom LaTeX package list, that includes those we need for Xournal++ specifically.

kenvandine commented on snapcraft.yaml:
```diff
@@ -35,7 +31,7 @@ apps:
     extensions: [gnome-3-38]
     environment:
-      PATH: $PATH:$SNAP/usr/bin:$SNAP/bin
+      PATH: $PATH:$SNAP/usr/bin:$SNAP/bin:$SNAP/usr/share/.TinyTeX/bin/x86_64-linux
```
Can we make this path arch independent?  Perhaps move it during the override-build step to be simply $SNAP/usr/share/.TinyTeX/bin/

rolandlo replied:
The trouble is that this directory contains relative links, which need to be regenerated when the directory is moved.

kenvandine replied:
Ok, perhaps just add both paths, it would be harmless

PATH: $PATH:$SNAP/usr/bin:$SNAP/bin:$SNAP/usr/share/.TinyTeX/bin/x86_64-linux:$SNAP/usr/share/.TinyTeX/bin/aarch64-linux

rolandlo replied:
I have also added the aarch64 path for Lua, so that the lgi module would be found on aarch64 as well.

kenvandine reviewed (changes requested):
One inline comment, otherwise looks good.  Thanks!

rolandlo committed:
add arch64 paths

kenvandine commented:
Looks great, nice improvement
